ENIGMAsystem/SHELL/Graphics_Systems/General/GSmatrix.cpp

```diff
@@ -95,6 +101,11 @@ bool is_vec4(const var& value) {
   return (value.array_len() == 4);
 }
 
+bool is_matrix(const var& value) {
+  return (value.array_len() == 16)
+      || (value.array_len() == 4 && value.array_height() == 4);
```

Comment:
If you want to make this "correct" now, I can tell you that matrices in GMS are 1D arrays of 16 values **_NOT_** 2D 4x4 arrays.

>This function takes a 16 value 1D array (which you can create using the matrix_build or matrix_get functions) and uses it to set one of the following three matrix types:
https://docs.yoyogames.com/source/dadiospice/002_reference/drawing/drawing%203d/3d%20transformations/matrix_set.html

>This function takes a 1D array of 16 values, corresponding to a given 4x4 matrix type, where elements [0 - 3] would be row 1, elements [4 -7] would be row 2, etc... (see the image on the main page). You can create such a matrix using the matrix_build or matrix_get functions or simply building the array yourself and passing that into the function.
https://docs2.yoyogames.com/source/_build/3_scripting/4_gml_reference/matrices/matrix_set.html

Reply:
Ok: 797ad2a399c7164c590a99d707c05e6b645e646c

I undid that for now, I'd rather just keep this simple so it's compatible for now, bigger fish to fry. Because of the coming GML language changes, we probably won't even have 2D arrays anyway.